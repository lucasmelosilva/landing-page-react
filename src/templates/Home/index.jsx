import { Heading } from "../../components/Heading"
import { theme } from "../../styles/theme"
import * as Styled from "./styles"

function Home() {
  return (
    <div className="App">
      <Styled.Wrapper>
        <Heading>Hello world</Heading>
      </Styled.Wrapper>
    </div>
  )
}

export default Home
