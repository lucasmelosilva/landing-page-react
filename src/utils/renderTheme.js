import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { theme } from "../styles/theme";

export const renderTheme = (children) => render(
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)