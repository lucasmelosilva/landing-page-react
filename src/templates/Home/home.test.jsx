import { screen } from '@testing-library/react'
import Home from '.'
import { renderTheme } from '../../utils/renderTheme'

test('render learn react link', () => {
  const {debug} = renderTheme(<Home />)
  const headingContainer = screen.getByRole('heading', { name: 'Hello'})
    .parentElement
  expect(headingContainer).toHaveStyle({
    background: 'blue'
  })
  expect(headingContainer).toMatchSnapshot()
})