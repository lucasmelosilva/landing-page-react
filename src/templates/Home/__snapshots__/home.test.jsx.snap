// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render learn react link 1`] = `
<div
  class="sc-beySPh crUDSY"
>
  <h1>
    Hello
  </h1>
</div>
`;
